[gradle-plugin]: Latest update (5.5.0) is broken on JDK loom eap builds

- [x] summary of problem
     This [commit](https://github.com/diffplug/spotless/commit/086c84326e3b2e1968ae432e6169e3f10e744618#diff-7a7322bbb00bcfb47ee56dbf3193672bR84) is broken on JDK loom EAP builds (`java.lang.StringIndexOutOfBoundsException`). The plugin was working fine on all JDK eap versions, including project [loom](http://jdk.java.net/loom/) . The version format used is, 

```
$ java -version    
openjdk version "16-loom" 2021-03-16
OpenJDK Runtime Environment (build 16-loom+6-105)
OpenJDK 64-Bit Server VM (build 16-loom+6-105, mixed mode, sharing)
```
Is it possible to handle the EAP version also?

- [x] gradle or maven version
    * `6.7-rc-1`
- [x] spotless version
   * `5.5.0`
- [x] operating system and version
   * Mac OSX (`10.13.6`)
- [x] copy-paste your full Spotless configuration block(s), and a link to a public git repo that reproduces the problem if possible
    
```
spotless {
    java {
        googleJavaFormat(Versions.googleJavaFormat)
    }
  kotlin {
        ktlint(Versions.ktlint).userData(mapOf("disabled_rules" to "no-wildcard-imports"))
        targetExclude("$buildDir/**/*.kt", "bin/**/*.kt")
        // licenseHeader(License.Apache)
    }
  kotlinGradle {
        ktlint(Versions.ktlint).userData(mapOf("disabled_rules" to "no-wildcard-imports"))
        target("*.gradle.kts")
    }
  format("misc") {
        target("**/*.md", "**/.gitignore")
        trimTrailingWhitespace()
        endWithNewline()
    }
 }
 ```
- [x] copy-paste the full content of any console errors emitted by `gradlew spotless[Apply/Check] --stacktrace`
```
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 7
        at com.diffplug.spotless.java.GoogleJavaFormatStep.<clinit>(GoogleJavaFormatStep.java:84)
        ... 199 more
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,8 @@ This document is intended for Spotless developers.
 We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (starting after version `1.27.0`).
 
 ## [Unreleased]
+### Fixed
+* Improved JRE parsing to handle strings like `16-loom` (fixes [#693](https://github.com/diffplug/spotless/issues/693)).
 
 ## [2.6.0] - 2020-09-11
 ### Added

--- a/lib/src/main/java/com/diffplug/spotless/java/GoogleJavaFormatStep.java
+++ b/lib/src/main/java/com/diffplug/spotless/java/GoogleJavaFormatStep.java
@@ -19,6 +19,8 @@ import java.io.IOException;
 import java.io.Serializable;
 import java.lang.reflect.Method;
 import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import com.diffplug.spotless.FormatterFunc;
 import com.diffplug.spotless.FormatterStep;
@@ -81,7 +83,14 @@ public class GoogleJavaFormatStep {
 		if (jre.startsWith("1.8")) {
 			JRE_VERSION = 8;
 		} else {
-			JRE_VERSION = Integer.parseInt(jre.substring(0, jre.indexOf('.')));
+			Matcher matcher = Pattern.compile("(\\d+)").matcher(jre);
+			if (!matcher.find()) {
+				throw new IllegalArgumentException("Expected " + jre + " to start with an integer");
+			}
+			JRE_VERSION = Integer.parseInt(matcher.group(1));
+			if (JRE_VERSION <= 8) {
+				throw new IllegalArgumentException("Expected " + jre + " to start with an integer greater than 8");
+			}
 		}
 	}
 

--- a/plugin-gradle/CHANGES.md
+++ b/plugin-gradle/CHANGES.md
@@ -3,6 +3,8 @@
 We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (starting after version `3.27.0`).
 
 ## [Unreleased]
+### Fixed
+* Improved JRE parsing to handle strings like `16-loom` (fixes [#693](https://github.com/diffplug/spotless/issues/693)).
 
 ## [5.5.0] - 2020-09-11
 ### Added

--- a/plugin-maven/CHANGES.md
+++ b/plugin-maven/CHANGES.md
@@ -3,6 +3,8 @@
 We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (starting after version `1.27.0`).
 
 ## [Unreleased]
+### Fixed
+* Improved JRE parsing to handle strings like `16-loom` (fixes [#693](https://github.com/diffplug/spotless/issues/693)).
 
 ## [2.3.0] - 2020-09-11
 ### Added

--- a/testlib/src/main/java/com/diffplug/spotless/JreVersion.java
+++ b/testlib/src/main/java/com/diffplug/spotless/JreVersion.java
@@ -16,19 +16,27 @@
 package com.diffplug.spotless;
 
 import java.util.function.IntPredicate;
-
-import com.diffplug.common.base.StandardSystemProperty;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class JreVersion {
 	private JreVersion() {}
 
 	/** Returns the major version of this VM, e.g. 8, 9, 10, 11, 13, etc. */
 	public static int thisVm() {
-		String jvmVersion = StandardSystemProperty.JAVA_VERSION.value();
-		if (jvmVersion.startsWith("1.8.")) {
+		String jre = System.getProperty("java.version");
+		if (jre.startsWith("1.8")) {
 			return 8;
 		} else {
-			return Integer.parseInt(jvmVersion.substring(0, jvmVersion.indexOf('.')));
+			Matcher matcher = Pattern.compile("(\\d+)").matcher(jre);
+			if (!matcher.find()) {
+				throw new IllegalArgumentException("Expected " + jre + " to start with an integer");
+			}
+			int version = Integer.parseInt(matcher.group(1));
+			if (version <= 8) {
+				throw new IllegalArgumentException("Expected " + jre + " to start with an integer greater than 8");
+			}
+			return version;
 		}
 	}
 

